### DISCUSSION TITLE
Configuration meta_client

### DISCUSSION THREAD

**Discussion by levcom** [GreptimeTeam/greptimedb#3791] [General]
As well as meta_client for standalone configuration (single process)

```
frontend: FrontendOptions {
        mode: Standalone,
...
        meta_client: None,
...
}
```

I wanted to connect MetaClient to my program and use DatafusionQueryEngine.
For example, in my program I want to execute several queries that return RecordBatches and perform join, projects ops,  mathematical ops with them (with DatafusionQueryEngine).

**Comment by killme2008** [comment]
Hi,

I am sorry that I don't get the point. Could you please provide a more detailed description of your question?

**Reply by levcom** [reply]
I want to execute multiple queries on a database to get some results. Then, I want to perform additional queries on these results for merging, filtering, and more. To do this, I plan to use the DatafusionQueryEngine for post-processing in my Rust program. But to create a DatafusionQueryEngine instance, I need to connect a MetaClient. However, the address for the MetaClient is not provided in the standalone configuration (it’s set to meta_client: None), and I’m not sure how to set it up.

`
        let grpc_addr = "127.0.0.1:4001";
        let client = Client::with_urls([grpc_addr]);
        let db = Database::new(DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME, client);

        let res1 = database.sql(&query1).await.unwrap();
        let res2 = database.sql(&query2).await.unwrap();

        // join res1 and res2 filter and project

        // meta_addr - undefinded !!!
        let mut meta_client = MetaClientBuilder::default().enable_store().build();
        meta_client.start([meta_addr]).await.unwrap();
        let meta_client = Arc::new(meta_client);
        ...
        let engine = DatafusionQueryEngine::new(state, _)
`

**Reply by killme2008** [reply]
If I understand correctly, you can set the meta client to `None` when in standalone mode.
